xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.h

```diff
@@ -42,6 +42,31 @@ struct VideoPicture
   VideoPicture& SetParams(const VideoPicture &pic);
   void Reset(); // reinitialize members, videoBuffer will be released if set!
 
+  /*!
+  \brief Checks if the video picture has A53 side data (closed captions).
+  *
+  * \return true if the video picture has A53 side data, false otherwise
+  */
+  bool HasA53SideData() const { return m_A53SideData != nullptr; }
+
+  /*!
+  \brief Gets the A53 side data of the video picture (closed captions)
+  * \return the A53 side data, reseting the value
+  */
+  AVBufferRef* ConsumeA53SideData()
+  {
+    AVBufferRef* sideData = m_A53SideData;
+    m_A53SideData = nullptr;
+    return sideData;
+  }
+
+  /*!
+  \brief Set the A53 side data on the video picture
+  *
+  * \param[in] sideData the data to store as A53 side data
+  */
+  void SetA53SideData(AVBufferRef* sideData) { m_A53SideData = sideData; }
```

Comment:
Should we be calling `av_buffer_ref` and `av_buffer_unref` to do the ref counting for the buffer?

Reply:
Yeah nice catch